corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -119,14 +121,26 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                 }
                 else {
                     // Set lookup table option by label
-                    var option = domElement.find("option").filter(function (_) {
-                        return $(this).text().trim() === value;
-                    });
-                    if (option.length === 1) {
-                        domElement.val(String(option.index() - 1)).trigger("change");
-                    } else {
-                        domElement.val("-1").trigger('change');
+                    var matchingOption = function (el) {
+                        return el.find("option").filter(function (_) {
+                            return $(this).text().trim() === value;
+                        });
                     }
+                    domElement.each(function () {
+                        var option = matchingOption($(this));
+                        if (this.multiple === true) {
+                            if (option.length === 1 && $(this).val().indexOf(option.val()) === -1) {
```

Comment:
Nitpick: the multiple expressions in this line and the one below are a bit hard to read. You could simplify this block a bit by just assigning a value to a local var, and then doing the `$(this).val(value).trigger("change")` afterwards.

Reply:
Sure

Reply:
Sure